### ISSUE TITLE
Make serializing generic query results better

### ISSUE THREAD

**Issue by aumayr** [aumayr/fava#168]
Exporting to CSV/XLS/XLSX means converting the results into a 2-dimensional array. It would be nice to convert columns of type `Cost` (details of which type a column is are already included in the result from the API) into two columns for number and currency. 

Origin: #166 by @alexjj

> Now that you've added exporting queries to csv/xls etc. I realised it would be more useful to separate number and currency into two columns. The number column is filled red and the type error appears. It doesn't really impact the functionality of the query, although when exporting to xls/ods it's not formatted as a number and it saves it as '12.34, which obviously means formula don't recognise it as a number.
> 
> For reference the query is:
> 
> ```sql
> SELECT 
>     month, account, sum(number), currency
> FROM
>     year = 2016
> WHERE 
>     account ~ 'Expenses'
> GROUP BY month, account, currency
> ORDER by month, currency, account
> FLATTEN
> ```

**Comment by alexjj** [comment]
I hope my original issue didn't confuse the two issues, and I'll reply to #166 shortly. The Query-error type not recognized is solely in fava web view. It may or may not impact how the spreadsheet columns are generated. I've used xlsx writer (I know you're not though) a _tiny_ amount in the past and remember specifying the [format](https://xlsxwriter.readthedocs.org/format.html#set_num_format) for a cell. This was the type of thing I had in mind as a possible solution. The [pyexcel docs](https://pyexcel.readthedocs.org/en/latest/tutorial03.html) weren't entirely detailed on formatters for me to be sure.

At home I use libreoffice, and opened the ODS. At work I have Excel 2010 and opened the .xls and .xlsx in that.

I use English-US (work) and English-UK (home).

The .ods I opened in libreoffice, the number column had them listed as '123.43 and to make it a number I'd have to remove the '.

For .xls and .xlsx, Excel sees the numbers as strings. It'll count the number of cells but won't do any calculations. It doesn't treat it as a number. The ' isn't there but might as well be. If I edit/activate the cell and hit enter it registers  as a number.

I hope that's a bit clearer

**Comment by aumayr** [comment]
Thanks @alexjj for clarifying, I understand the problem now and will have a look at the formatters in pyexcel. 

Could you attach the ODS and XLS-files to this issue (if they do not contain any personal info)? This way I could test it on exactly those files. 

(Sidenote: This feature will probably end up in beancount, because these reports should be generated there. But if we fix issues like this now, the results will be better when we port this over to beancount.)

**Comment by alexjj** [comment]
Here's a sample of what it outputs. All I did was delete the account names.
[Monthly_Totals_-_2016.xlsx](https://github.com/aumayr/fava/files/149516/Monthly_Totals_-_2016.xlsx)

**Comment by corani** [comment]
A quick-fix that doesn't turn integers and decimals into strings in #189 

@aumayr I agree it makes sense to automatically split Positions and Inventory into two columns.

**Comment by corani** [comment]
I pushed some more changes to automatically split `Position` and `Inventory`. It works, but it's not the most elegant code. @aumayr or @yagebu feel free to clean it up before merging.

**Comment by yagebu** [comment]
Is there anything more to be done here after #168 @alexjj?
